src/vmm/src/config.rs

```diff
@@ -1,13 +1,100 @@
 // Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0 OR BSD-3-Clause
 
+use std::collections::HashMap;
 use std::convert::TryFrom;
 use std::fmt;
+use std::num;
 use std::path::PathBuf;
 use std::result;
 
 use super::{DEFAULT_HIGH_RAM_START, DEFAULT_KERNEL_CMDLINE};
 
+trait ArgParsing {
+    fn parse(param_name: &'static str, raw: String) -> Result<Self, ParsingError>
+    where
+        Self: Sized;
+}
+
+macro_rules! argparsing{
+    ($($t:ty)*) => ($(impl ArgParsing for $t {
+        fn parse(param_name: &'static str, raw: String) -> Result<Self, ParsingError> {
+            raw.parse().map_err(|err: num::ParseIntError| ParsingError::ParsingFailed(param_name, err.to_string()))
+        }
+    })*)
+}
+
+argparsing! { u32 u64 num::NonZeroU8 }
+
+impl ArgParsing for PathBuf {
+    fn parse(_param_name: &'static str, raw: String) -> Result<Self, ParsingError> {
+        Ok(PathBuf::from(raw))
+    }
+}
+impl ArgParsing for String {
+    fn parse(_param_name: &'static str, raw: String) -> Result<Self, ParsingError> {
+        Ok(raw)
+    }
+}
+
+fn tokenize(input: String) -> HashMap<String, String> {
+    input
+        .split(',')
+        .filter(|tok| !tok.is_empty())
+        .map(|tok| {
+            let mut iter = tok.split('=');
+            let param_name = iter.next().unwrap();
+            let value = iter.collect::<Vec<&str>>().join("=");
+            (param_name.to_lowercase(), value)
+        })
```

Comment:
I believe this logic is to be able to handle something like `cmdlin=\"foo=bar\"`? Is that right? How about something like the following - because this code is a bit difficult to read. 

```rust 

.map(|tok| {
            let mut iter = tok.splitn(2, '=');
            let param_name = iter.next().unwrap();
            let value = iter.next().unwrap().to_string();
            (param_name.to_lowercase(), value)
        })
```

I quickly tried this out in [playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=4c5c8b81a792954edfec9c45aa33eca8)  and seems to work. If this is what you are expecting.